client/src/app/shared/overload-js-functions.ts

```diff
@@ -117,6 +123,13 @@ function overloadArrayFunctions(): void {
         },
         enumerable: false
     });
+
+    Object.defineProperty(Array.prototype, 'tap', {
+        value: function <T>(callbackFn: (self: T[]) => T[]): T[] {
+            return callbackFn(this);
+        },
```

Comment:
```suggestion
        value: function <T>(callbackFn: (self: T[]) => void): T[] {
            callbackFn(this);
            return this;
        },
```

Currently, if you tap callback returns something else, it will replace the list. But you just want tapping without changing the array itself.